src/main/java/de/zalando/aruha/nakadi/controller/EventStreamController.java

```diff
@@ -0,0 +1,167 @@
+                            "stream_limit can't be lower than batch_limit");
+                    return;
+                }
+                if (streamTimeout != 0 && streamTimeout < batchTimeout) {
+                    writeProblemResponse(response, outputStream, UNPROCESSABLE_ENTITY,
+                            "stream_timeout can't be lower than batch_flush_timeout");
+                    return;
+                }
+
+                // deserialize cursors
+                List<Cursor> cursors = null;
+                if (cursorsStr != null) {
+                    try {
+                        cursors = jsonMapper.<List<Cursor>>readValue(cursorsStr,
+                                new TypeReference<ArrayList<Cursor>>() {
+                                });
+                    } catch (IOException e) {
+                        writeProblemResponse(response, outputStream, BAD_REQUEST,
+                                "incorrect syntax of X-nakadi-cursors header");
+                        return;
+                    }
+                }
+
+                // check that offsets are not out of bounds and partitions exist
+                if (cursors != null && !topicRepository.areCursorsValid(topic, cursors)) {
+                    writeProblemResponse(response, outputStream, PRECONDITION_FAILED, "cursors are not valid");
+                    return;
+                }
+
+                // if no cursors provided - read from the newest available events
+                if (cursors == null) {
+                    cursors = topicRepository
+                            .listPartitions(topic)
+                            .stream()
+                            .map(pInfo -> new Cursor(pInfo.getPartitionId(), pInfo.getNewestAvailableOffset()))
+                            .collect(Collectors.toList());
+                }
+
+                final Map<String, String> streamCursors = cursors
+                        .stream()
+                        .collect(Collectors.toMap(
+                                Cursor::getPartition,
+                                Cursor::getOffset));
+
+                final EventStreamConfig streamConfig = new EventStreamConfig(topic, streamCursors, batchLimit,
+                        streamLimit, batchTimeout, streamTimeout, streamKeepAliveLimit);
+
+                response.setStatus(HttpStatus.OK.value());
+
+                final String acceptEncoding = request.getHeader("Accept-Encoding");
+                final boolean gzipEnabled = acceptEncoding != null && acceptEncoding.contains("gzip");
+                final OutputStream output = gzipEnabled ? new FlushableGZIPOutputStream(outputStream) : outputStream;
+
+                if (gzipEnabled) {
+                    response.addHeader("Content-Encoding", "gzip");
+                }
+
+                final EventConsumer eventConsumer = topicRepository.createEventConsumer(topic,
+                        streamConfig.getCursors());
```

Comment:
I don't think so. Because in `areCursorsValid` you fetch all available topic and search for each of it in the cursors list provided by the user - not the other way around. If a user specifies `foobar` as partition id, the offset won't even validated at all in 106.